sortinghat/parsing/grimoirelab.py

```diff
@@ -0,0 +1,358 @@
+
+        - organizations:
+            Bitergia:
+                - bitergia.com
+                - support.bitergia.com
+                - biterg.io
+            LibreSoft:
+                - libresoft.es
+
+        :param json: YAML object to parse
+
+        :raises InvalidFormatError: raised when the format of the YAML is
+            not valid.
+        """
+        if not stream:
+            return
+
+        yaml = self.__load_yml(stream)
+
+        try:
+            for element in yaml:
+                name = self.__encode(element['organization'])
+                o = Organization(name=name)
+
+                if 'domains' in element:
+                    if isinstance(element['domains'], list):
+                        for dom in element['domains']:
+                            d = Domain(domain=dom, is_top_domain=False)
+                            o.domains.append(d)
+                    else:
+                        msg = "invalid GrimoireLab yaml format. List of elements expected"
+                        raise InvalidFormatError(cause=msg)
+                self._organizations[name] = o
+
+        except KeyError as e:
+            msg = "invalid GrimoireLab yaml format. Attribute %s not found" % e.args
+            raise InvalidFormatError(cause=msg)
+        except TypeError as e:
+            msg = "invalid GrimoireLab yaml format. %s" % e.args
+            raise InvalidFormatError(cause=msg)
+
+    def __parse_affiliations_yml(self, affiliations, uuid):
+        """Parse identity's affiliations from a yaml dict.
+
+        The organizations are previously analyzed so here we create the
+        link for just the ones with an entry in the organizations file.
+        """
+
+        enrollments = []
+
+        for aff in affiliations:
+
+            name = self.__encode(aff['organization'])
+            org = self._organizations.get(name, None)
+
+            if org == None:
+                continue
+
+            if 'start' in aff:
+                start_date = self.__force_datetime(aff['start'])
+            else:
+                start_date = MIN_PERIOD_DATE
+
+            if 'end' in aff:
+                end_date = self.__force_datetime(aff['end'])
+            else:
+                end_date = MAX_PERIOD_DATE
+
+            enrollment = Enrollment(start=start_date, end=end_date,
+                                    organization=org)
+            enrollments.append(enrollment)
+
+        self.__validate_enrollment_periods(enrollments)
+
+        return enrollments
+
+    def __force_datetime(self, obj):
+        """Converts ojb to datetime.datetime
+
+        YAML parsing returns either date or datetime object depending
+        on how the date is written. YYYY-MM-DD will return a date and
+        YYYY-MM-DDThh:mm:ss will return a datetime
+
+        :param obj: date or datetime object
+        """
+        if isinstance(obj,datetime):
+            return obj
+
+        from datetime import time
```

Comment:
This should be imported at the beginning of the code. Don't use `from ... import ...` statements if possible.